### PR TITLE
IPP 12 - Add Rule and RuleResult to Swagger

### PR DESCRIPTION
**Pull request by dLobatog**
Depends on : 
https://github.com/RedHatInsights/compliance-backend/pull/94/files
https://github.com/RedHatInsights/compliance-backend/pull/93/files

### CODE REVIEW

File: app/controllers/profiles_controller.rb [hunk 1]
```diff
@@ -18,4 +15,10 @@ def show
     authorize profile
     render json: ProfileSerializer.new(profile)
   end
+
+  private
+
+  def resource
+    Profile
```
Comment (akofink) [inline app/controllers/profiles_controller.rb]: I'm a bit confused. Are you using this method? And why not put these changes in the "IPP 12 - Change per-page/page to limit/offset and add links" commit where you add the Metadata and Pagination modules?

File: spec/swagger_helper.rb [hunk 2]
```diff
@@ -81,6 +81,56 @@
...
+            ref_id: {
+              type: 'string',
+              example: 'xccdf_org.ssgproject.content_rule_directory_access_'\
+              'var_log_audit'
+            },
+            severity: {
+              type: 'string',
+              example: 'Low'
+            },
+            description: {
```
Comment (akofink) [inline spec/swagger_helper.rb]: Where is this stuff coming from? This file isn't autogenerated, is it?

File: swagger/v1/openapi.json [hunk 3]
```diff
@@ -167,6 +167,257 @@
         }
       }
     },
+    "/api/compliance/rule_results": {
```
Comment (akofink) [inline swagger/v1/openapi.json]: This _is_ autogenerated, right? Do we have instructions in the readme for how to generate the openapi docs?

File: test/controllers/systems_controller_test.rb [hunk 4]
```diff
@@ -5,20 +5,17 @@
 class SystemsControllerTest < ActionDispatch::IntegrationTest
   setup do
     SystemsController.any_instance.expects(:authenticate_user)
+    SystemsController.any_instance.expects(:policy_scope).with(Host)
```
Comment (akofink) [inline test/controllers/systems_controller_test.rb]: This isn't going to work here once we add tests for `show`. You can put it in a method, though, if you want it more DRY.

File: spec/swagger_helper.rb (hunk 2 above)
Comment (dLobatog) [reply spec/swagger_helper.rb]: It's not autogenerated, this is the OpenAPI definition I wrote

File: swagger/v1/openapi.json (hunk 3 above)
Comment (dLobatog) [reply swagger/v1/openapi.json]: Correct: I use this: https://github.com/domaindrivendev/rswag `rake rswag:specs:swaggerize`

File: test/controllers/systems_controller_test.rb (hunk 4 above)
Comment (dLobatog) [reply test/controllers/systems_controller_test.rb]: Done, I've kept it separate for now

File: swagger/v1/openapi.json (hunk 3 above)
Comment (akofink) [reply swagger/v1/openapi.json]: Do you think it'd be a good idea to mention that somewhere in the Readme?

**Comment by dLobatog** [comment]
https://github.com/RedHatInsights/compliance-backend/pull/97

**Pull request merged**